src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -112,141 +107,8 @@ private[snowflake] class JDBCWrapper {
-    params.setJDBCProxyIfNecessary(jdbcProperties)
-
-    // Adding Authenticator parameter
-    params.sfAuthenticator match {
-      case Some(value) =>
-        jdbcProperties.put("authenticator", value)
-      case _ => // No default value for it.
-    }
-
-    // Always set CLIENT_SESSION_KEEP_ALIVE.
-    // Note, can be overridden with options
-    jdbcProperties.put("client_session_keep_alive", "true")
-
-    // Force DECIMAL for NUMBER (SNOW-33227)
-    if (params.treatDecimalAsLong) {
-      jdbcProperties.put("JDBC_TREAT_DECIMAL_AS_INT", "true")
-    } else {
-      jdbcProperties.put("JDBC_TREAT_DECIMAL_AS_INT", "false")
-    }
-
-    // Add extra properties from sfOptions
-    val extraOptions = params.sfExtraOptions
-    for ((k: String, v: Object) <- extraOptions) {
-      jdbcProperties.put(k.toLowerCase, v.toString)
-    }
-
-    // Only one JDBC version is certified for each spark connector.
-    if (!Utils.CERTIFIED_JDBC_VERSION.equals(Utils.jdbcVersion)) {
-      log.warn(
-        s"""JDBC ${Utils.jdbcVersion} is being used.
-           | But the certified JDBC version
-           | ${Utils.CERTIFIED_JDBC_VERSION} is recommended.
-           |""".stripMargin.filter(_ >= ' '))
-    }
-
-    // Important: Set client_info is very important!
-    // For more details, refer to PROPERTY_NAME_OF_CONNECTOR_VERSION
-    // NOTE: From JDBC 3.13.3, the client info can be set with JDBC properties
-    // instead of system property: "snowflake.client.info".
-    jdbcProperties.put(SFSessionProperty.CLIENT_INFO.getPropertyKey, Utils.getClientInfoString())
-
-    val conn: Connection = DriverManager.getConnection(jdbcURL, jdbcProperties)
-
-    // Setup query result format explicitly because this option is not supported
-    // to be set with JDBC properties
-    if (jdbcProperties.getProperty(Parameters.PARAM_JDBC_QUERY_RESULT_FORMAT) != null) {
-      try {
-        val resultFormat =
-          jdbcProperties.getProperty(Parameters.PARAM_JDBC_QUERY_RESULT_FORMAT)
-        conn
-          .createStatement()
-          .execute(
-            s"alter session set JDBC_QUERY_RESULT_FORMAT = '$resultFormat'"
-          )
-      } catch {
-        case e: SQLException =>
-          log.info(e.getMessage)
-        case other: Any =>
-          // Rethrow other errors.
-          throw other
-      }
-    }
-
-    // Setup query result format explicitly because this option is not supported
-    // to be set with JDBC properties
-    if (params.supportAWSStageEndPoint) {
-      params.getS3StageVpceDnsName.map {
-        x => conn.createStatement().execute(s"alter session set S3_STAGE_VPCE_DNS_NAME = '$x'")
-      }
-    }
-
-    // Send client info telemetry message.
-    val extraValues = Map(TelemetryClientInfoFields.SFURL -> sfURL)
-    SnowflakeTelemetry.sendClientInfoTelemetry(extraValues, conn)
-
-    conn
-  }
+  def getConnector(params: MergedParameters): ServerConnection =
+    ServerConnection.getServerConnection(params)
```

Comment:
The actual JDBC connection creation is moved to `ServerConnection.createJDBCConnection()`